server/test/api/accounts.js

```diff
@@ -121,6 +122,28 @@ describe('The account API, PATH : /accounts', function () {
       .catch(helpers.handler);
   });
 
+ it('METHOD : GET, PATH : /accounts/:id/balance, It returns an error 404 not found', function () {
+   return agent.get('/accounts/:id/balance'.replace(':id', FETCHABLE_ACCOUNT_ID))
+    .then(function (res){
+      expect(res).to.have.status(404);
+      expect(res).to.be.json;
+    })
+    .catch(helpers.handler);
+ });
+
+  
+  it('METHOD : GET, PATH : /accounts/:id/balance?journal=1, It returns the balance of a provided account_id, scans the journal also', function () {
+    return agent.get('/accounts/:id/balance?journal=1'.replace(':id', ACCOUNT_ID_FOR_BALANCE))
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res).to.be.json;
+        expect(res.body).to.not.be.empty;
+        expect(res.body).to.have.all.keys('account_id', 'debit', 'credit', 'balance');
+        expect(res.body.debit).to.satisfy(function (debit) { return debit >= 0;});
```

Comment:
It's a good thought, but I think testing the `IFNULL()` function is also not necessary.

A better test would be to know exactly the account balance in question and do this:
```js
expect(res.body.debit).to.equal(0);
expect(res.body.credit).to.equal(12.35);
```
or something similar.